src/model/neuroevolution/individual.py

```diff
@@ -27,35 +27,77 @@ def get_random_layer_index(self) -> int:
     def get_random_weight_index(neural_network_layer: Layer) -> Tuple[int, ...]:
         return tuple(np.random.randint(neural_network_layer.weights.shape))
 
+    @staticmethod
+    def get_random_weight_indexes(
+        neural_network_layer: Layer, indexes_count: int
+    ) -> Tuple[Tuple[int, ...], ...]:
+        return tuple(
+            np.random.randint(neural_network_layer.weights.shape, size=indexes_count)
+        )
+
```

Comment:
This code doesn't even work (it doesn't crash for `count=1`, but it doesn't do what you might expect - this code returns a list with one `int`, which is later used by `put`, which operates only on flat arrays, which hides this bug).
Other than that (if this code did what was expected) you are returning a tuple of `ndarray`s, not a tuple of tuples, `mypy` just can't see that as it doesn't have definitions for `numpy`.

```suggestion
    def get_random_weight_indexes(
        layer: Layer, indexes_count: int
    ) -> Tuple[np.ndarray, ...]:
        """
        Calculates indexes_count random weight indexes for a single layer.
        
        Returned tuple is ready to be used immediately for indexing the layer's weights.
        """
        weights = layer.weights
        return tuple(
            np.random.randint(0, weights.shape, size=(indexes_count, weights.ndim)).T
        )
```